test/test-file.cpp

```diff
@@ -46,38 +46,41 @@ namespace quick_lint_js {
 namespace {
 filesystem::path make_temporary_directory();
 void write_file(filesystem::path, const std::string &content);
+filesystem::path temp_file_path;
 
 class test_file : public ::testing::Test {
  public:
   filesystem::path make_temporary_directory() {
-    // TODO(strager): Delete the directory when the test finishes.
     return quick_lint_js::make_temporary_directory();
   }
+
+ protected:
+  void TearDown() override { filesystem::remove_all(temp_file_path); }
```

Comment:
Almost-blocking: What happens if `filesystem::remove_all` is given an empty path? (A call with an empty path happens if a test doesn't assign to `temp_file_path`.) I think we should only call `filesystem::remove_all` if we create a directory. Perhaps we could use [`std::optional`](https://en.cppreference.com/w/cpp/utility/optional) or [`std::vector`](https://en.cppreference.com/w/cpp/container/vector) for this?